plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/condition/ShellHasChildrenOrIsNotAvailable.java

```diff
@@ -26,34 +30,45 @@
 	
 	private Shell shell;
 	private static final Logger log = Logger.getLogger(ShellHasChildrenOrIsNotAvailable.class);
+	private List<org.eclipse.swt.widgets.Shell> resultChildren;
 
 	/**
 	 * Default constructor.
 	 * @param deleteShell instance of shell to test
 	 */
 	public ShellHasChildrenOrIsNotAvailable(Shell deleteShell) {
 		this.shell = deleteShell;
+		this.resultChildren = new ArrayList<>();
 	}
 
 	@Override
 	public boolean test() {
-		int childShells = 0;
 		org.eclipse.swt.widgets.Shell swtShell = shell.getSWTWidget();
 		try {
-			childShells = ShellLookup.getInstance().getShells(swtShell).length;
+			this.resultChildren = Arrays.asList(ShellLookup.getInstance().getShells(swtShell));
 		} catch (RedDeerException e) {
 			if(swtShell != null){
 				return WidgetHandler.getInstance().isDisposed(swtShell);
 			}
 			return true;
 		}
-		log.debug("number of child shells: " + childShells);
-		return childShells > 0;
+		log.debug("number of child shells: " + this.resultChildren.size());
+		if (this.resultChildren.size() > 0) {
+			return true;
+		}
+		this.resultChildren.clear();
```

Comment:
not necessary. If there was an element in the list, true was returned 2 lines above. If there was not any element, list is empty and clear is not necessary.